apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -1346,3 +1349,98 @@ func (a *ProvisionerAPI) CACert() (params.BytesResult, error) {
 	caCert, _ := cfg.CACert()
 	return params.BytesResult{Result: []byte(caCert)}, nil
 }
+
+func (p *ProvisionerAPI) ProfileChangeInfo(machines params.Entities) (params.ProfileChangeResults, error) {
+	results := make([]params.ProfileChangeResult, len(machines.Entities))
+	canAccess, err := p.getAuthFunc()
+	if err != nil {
+		logger.Errorf("failed to get an authorisation function: %v", err)
+		return params.ProfileChangeResults{}, errors.Trace(err)
+	}
+	for i, machine := range machines.Entities {
+		result, err := p.machineProfileChangeInfo(machine.Tag, canAccess)
+		if err != nil {
+			results[i].Error = common.ServerError(err)
+			continue
+		}
+		results[i] = result
+	}
+	return params.ProfileChangeResults{Results: results}, nil
+}
+
+func (p *ProvisionerAPI) machineProfileChangeInfo(machineTag string, canAccess common.AuthFunc) (params.ProfileChangeResult, error) {
+	mTag, err := names.ParseMachineTag(machineTag)
+	if err != nil {
+		return params.ProfileChangeResult{}, errors.Trace(err)
+	}
+	machine, err := p.getMachine(canAccess, mTag)
+	if err != nil {
+		return params.ProfileChangeResult{}, errors.Trace(err)
+	}
+	appName := machine.UpgradeCharmProfileApplication()
+	if appName == "" {
+		return params.ProfileChangeResult{}, errors.Trace(errors.New("no appname for profile charm upgrade"))
+	}
+	profiles, err := machine.CharmProfiles()
+	if err != nil {
+		return params.ProfileChangeResult{}, errors.Trace(err)
+	}
+	var oldProfileName string
+	for _, p := range profiles {
+		rev, err := lxdprofile.ProfileRevision(p)
+		if err != nil {
+			// TODO should these be errors?  when does provisioner get restarted?
+			return params.ProfileChangeResult{}, errors.Trace(err)
+		}
+		if strings.HasSuffix(p, fmt.Sprintf("%s-%d", appName, rev)) {
+			oldProfileName = p
+			break
+		}
+	}
+	if oldProfileName == "" {
+		// TODO - new profile for charm?
+	}
+	url := machine.UpgradeCharmProfileCharmURL()
+	if url == "" {
+		return params.ProfileChangeResult{}, errors.Trace(errors.New("no url for profile charm upgrade"))
+	}
+	chURL, err := charm.ParseURL(url)
+	if err != nil {
+		return params.ProfileChangeResult{}, errors.Trace(err)
+	}
+	ch, err := p.st.Charm(chURL)
+	if err != nil {
+		return params.ProfileChangeResult{}, errors.Trace(err)
+	}
+	return params.ProfileChangeResult{
+		AppName:        appName,
+		OldProfileName: oldProfileName,
+		CharmRev:       ch.Revision(),
+		Profile:        params.CharmLXDProfile(*ch.LXDProfile()),
```

Comment:
`ch.LXDProfile()` might be `nil` -> not sure what happens for a type conversion in go.